packages/@react-spectrum/table/src/TableView.tsx

```diff
@@ -417,6 +431,16 @@ function TableColumnHeader({column}) {
   }, state, ref);
 
   let columnProps = column.props as SpectrumColumnProps<unknown>;
+
+  // Can these be enforced with prop type validation?
+  if (columnProps.width && columnProps.allowsResizing) {
+    throw new Error('Controlled state is not yet supported with column resizing. Please use defaultWidth for uncontrolled column resizing or remove the allowsResizing prop.');
```

Comment:
So if someone does this it will not work as expected, the entire component may as well be broken. I think my main concern with moving this to a `.warn()` is that in order to utilize `allowsResizing` people will have to change their API from their current `width` prop to `defaultWidth`. Most devs will probably not realize this distinction unless they're forced to. 

If you think a warning in this situation is enough, then I can change it to a warning, but just know that the component will not work as a dev intends if they go down this route.